src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMeteringController.java

```diff
@@ -145,4 +106,159 @@ public void collectOpenshiftMetrics(String account, OffsetDateTime start, Offset
         });
     }
 
+    private Map<LabelSetKey, Map<String, String>> getSubscriptionLabels(String account,
+        OffsetDateTime start, OffsetDateTime end) {
+        OpenshiftMetricProperties openshiftProperties = metricProperties.getOpenshift();
+        Map<LabelSetKey, Map<String, String>> labelData = new HashMap<>();
+
+        QueryResult labelMetrics = prometheusService.runRangeQuery(
+            String.format(openshiftProperties.getSubscriptionLabelPromQL(), account),
+            start,
+            end,
+            openshiftProperties.getStep(),
+            openshiftProperties.getQueryTimeout()
+        );
+
+        if (StatusType.ERROR.equals(labelMetrics.getStatus())) {
+            throw new MeteringException(
+                String.format("Unable to fetch openshift metrics: %s", labelMetrics.getError())
+            );
+        }
+
+        labelMetrics.getData().getResult().forEach(r -> {
+            // Just need the label values from here.
+            Map<String, String> labels = r.getMetric();
+            String clusterId = labels.getOrDefault("_id", "");
+
+            r.getValues().forEach(measurement -> {
+                BigDecimal time = measurement.get(0);
+                labelData.put(new LabelSetKey(clusterId, time), labels);
+            });
+        });
+
+        return labelData;
+    }
+
+    private int generateEventsFromMetrics(Map<LabelSetKey, Map<String, String>> labelSets,
+        String account, OffsetDateTime start, OffsetDateTime end) {
+        OpenshiftMetricProperties openshiftProperties = metricProperties.getOpenshift();
+        log.info("Collecting metrics!");
```

Comment:
aha! RE: my other comment around line 96/97 about the log statement being paired with this method